config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/BaseConfigGenerator.java

```diff
@@ -56,10 +57,52 @@ public abstract class BaseConfigGenerator {
 
    protected String jbossHome;
    protected boolean forkEmbedded;
+   protected boolean resetSystemProperties;
+
    protected Path script;
    protected PrintWriter scriptWriter;
    protected StringBuilder scriptBuf;
 
+   protected Set<String> resetProperties = Set.of("java.version",
```

Comment:
Can't we reverse and only list the system properties to reset? This seems more robust to JVM changes. This set should be shared between feature-spec and config generation.

Reply:
That is another option but it would need to know exactly what Sys Prop can affect the embedded server, there are several properties that can be set by the user and can influence the embedded server. Since the setting is out of our control, I added the most restrictive set. 

We could reverse it, but that could be more fragile in the future since any new property that can influence the startup of the embedded server would need to be into account. Another option to complement this is to allow the user to specify with another plugin option, which Sys Prop from the current environment doesn't want to reset, basically, allow the user to extend this resetProperties set.

Do you still prefer to reverse the list? I think I could find the exact Sys Prop but it is more difficult to be sure what other ones can affect...I'm more thinking of any kind of user property that can even influence the XML parsers or factories used by the embedded server

Reply:
As an example of the fragility, notice we discovered accidentally one "javax.management.builder.initial"

Reply:
@yersan , I think that we control wildfly and know the set of system properties that could affect the provisioning. We don't control the JVM system properties. I would vote for reversing and making this property a list of properties a user could extend. Having an empty list would mean to just reset the set we know.

Reply:
> We don't control the JVM system properties

We don't change them when we are building WildFly, notice we generally use the default one when we are building the project.

But, ok, I can look at what could be the initial set to remove, although I'm more confident in using a more restrictive set.

Reply:
> This set should be shared between feature-spec and config generation.

@jfdenise Right, I did not find the correct place to place it, I'll do another review, do you have any place in mind?

Reply:
We could have it there: ForkedEmbeddedUtil That is in galleon-plugins, so should be shared by both config and feature-spec.